docs/configuration.md

```diff
@@ -190,6 +190,8 @@ of the most common options to set are:
     and it is up to the application to avoid exceeding the overhead memory space
     shared with other non-JVM processes. When PySpark is run in YARN or Kubernetes, this memory
     is added to executor resource requests.
+
+    NOTE: This configuration is not supported on Windows.
```

Comment:
This is a little misleading since on Windows the extra memory will still be added to the resource requests. It's just the process-level memory limit that is not implemented.

Reply:
I agree. A better note would be something like "Python memory usage may not be limited on platforms that do not support resource limiting, such as Windows".

Reply:
I would say it's just supported or unsupported tho rather than making it complicated. I'm not 100% sure how it works on Windows with other modes. For instance, extra memory will probably allocated via Yarn but looks no one tested it before. Technically, it does not work on local mode as well.